corfudb-tools/src/main/java/org/corfudb/compactor/CorfuStoreCompactorMain.java

```diff
@@ -102,14 +104,13 @@ private void checkpoint() {
                 .isClient(false)
                 .build(), corfuStore, compactorMetadataTables);
         try {
-            int retryCheckpointing = 5;
-            for (int i = 0; i < retryCheckpointing; i++) {
+            for (int i = 0; i < RETRY_CHECKPOINTING; i++) {
                 if (distributedCheckpointerHelper.hasCompactionStarted()) {
                     distributedCheckpointer.checkpointTables();
                     break;
                 }
                 log.info("Compaction cycle hasn't started yet...");
-                TimeUnit.SECONDS.sleep(1);
+                TimeUnit.SECONDS.sleep(30);
```

Comment:
Make it a field `RETRY_CHECKPOINTING_SLEEP_SECOND`? 
30s seems a bit too long. What is the consideration here? Is it to tolerate a layout change?

Reply:
Yes and also during upgrade scenarios when we call this method right after an instantTrigger, we might have to wait for a while for the server to instantiate compaction. The CompactorService calls the shouldTrigger method every 10s and so 5s should be good enough here.